### PR TITLE
Introduce shop state query

### LINKED ISSUE #251: Add query for ShopState

### PR DESCRIPTION
**Pull request by ipdae**
resolve #251 

```graphql
query {
  stateQuery {
    shop {
      address
      products {
        sellerAgentAddress
        sellerAvatarAddress
        productId
        price
        itemUsable {
          itemId
          itemType
          itemSubType
        }
        costume {
          itemId
          itemType
          itemSubType
        }
      }
    }
  }
}
```

### CODE REVIEW

**Review by area363 — APPROVED** [review]

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs [hunk 1]
```diff
@@ -556,6 +558,268 @@ public async Task GetTx()
...
+
+        [Fact]
+        public async Task StateQueryRankingMapState()
+        {
+            var userPrivateKey = new PrivateKey();
+            var userAddress = userPrivateKey.PublicKey.ToAddress();
+            var avatarAddress = userAddress.Derive(string.Format(CultureInfo.InvariantCulture,
+                CreateAvatar2.DeriveFormat, 0));
+            var service = MakeMineChroniclesNodeService(userPrivateKey);
+            StandaloneContextFx.NineChroniclesNodeService = service;
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs:626]: In my thought, `GraphType` test doesn't need `NineChroniclesNodeService`. how about using `BlockChain<T>` directly?

File: NineChronicles.Headless/GraphTypes/States/Models/Item/ItemUsableType.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using GraphQL.Types;
+using Nekoyume.Model.Item;
+
+namespace NineChronicles.Headless.GraphTypes.States.Models.Item
+{
+    public class ItemUsableType : ItemBaseType<ItemUsable>
+    {
+        public ItemUsableType() : base()
+        {
+            Field<NonNullGraphType<GuidGraphType>>(nameof(Consumable.ItemId));
```
[the remark this answers is not in the document]
Comment (ipdae) [reply NineChronicles.Headless/GraphTypes/States/Models/Item/ItemUsableType.cs:10]: 😅 my mistake.

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs (hunk 1 above)
Comment (ipdae) [reply NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs:626]: i think using `MakeMineChroniclesNodeService` is more convenient than using `BlockChain<T>`. (for testing config, make genesis, etc)

Comment (moreal) [reply NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs:626]: I agree with your opinion but there are some lines not needed for the purpose and it seems able to be resolved next pull request.

**Review by moreal — APPROVED** [review]

**Pull request merged**